### PR TITLE
Add Pov-3d viewer

### PR DESCRIPTION
**Pull request by JOHNKIM-KK**
## Description

- Project initial setting
- Create Viewer
- pov-3d Event
- Add Example

<br/>

## Project initial setting
- create vite
- Configure package.json

## Viewer
`<pov-3d-viewer>`
I have created the Pov_3d_viewer class. This class inherits from HTMLElement and defines a web component. It provides key features of a 3D viewer, including loading, rendering, and playing animations of 3D models. To use the Pov_3d_viewer class with the pov-3d-viewer tag, I registered the web component using the customElements.define method. This allows the insertion of the 3D viewer into an HTML document using the pov-3d-viewer tag.

Pov_3d_viewer Class: This class defines a web component by inheriting from HTMLElement. The constructor of the class performs the basic setup of the 3D viewer, and the load method is used to load a 3D model, while the render method is used to render the 3D model.
customElements.define: This method registers the Pov_3d_viewer class to be used with the pov-3d-viewer tag as a web component. After calling this method, the 3D viewer can be inserted into an HTML document using the pov-3d-viewer tag.

`preset`
In the attributeChangedCallback method of the Pov_3d_viewer class, you can detect changes to the preset attribute and add code to update the viewer's settings accordingly. When the value of the preset attribute changes, call the corresponding method of the ViewerOption class to update the viewer's settings.

### Attributes
The `pov-3d-viewer` custom element supports the following attributes:
- `model` - The path to the 3D model file.
- `preset` - The preset configuration to use.
  - Available presets: `Initial(Default)`, `Bright`, `Dark`, `Polyground`.
- `baseColor` - The base color of the 3D model.
  - Accepts hex color codes (e.g., `#ff0000`). 
- `autoRotate` - Whether to enable auto-rotation of the 3D model.
- `loadProgress` - Whether to display a loading progress bar.

### Events
The `pov-3d-viewer` custom element emits the following events:
- `pov-setup` - Emitted when the default environment setup is completed for the pov-3d-viewer.
- `pov-ready` - Emitted when the 3D model is ready to be loaded.
- `pov-event` - Emitted when a custom event is triggered.
  - `load-progress` - Emitted when the 3D model loading progress changes.
  ```javascript
  this.dispatchEvent(
      new CustomEvent("pov-event", {
        detail: {
          type: "load-progress",
          loaded: progressEvent.loaded,
          total: progressEvent.total,
        },
      }),
    );
  ```
- `pov-model-load` - Emitted when the 3D model is loaded.

### simple
Add an index.html file to the simple directory to provide basic viewer functionality.

It provides an environment for easy testing and demonstration of the default features of pov-3d-viewer. Users can directly experience the basic functionality of pov-3d-viewer.

### dynamic

**Attribute Setting**
By using the events emitted by the pov-3d-viewer custom element, control the attribute setting of the Pov-3d-Viewer to execute sequentially.

observedAttributes was used to define the attributes that pov-3d-viewer will observe. This enables the detection of changes in the asset, preset, and base_color attributes.
The attributeChangedCallback method was implemented to perform appropriate actions when attribute values change. This method receives the name of the attribute, its previous value, and its new value as arguments, allowing for the definition of actions based on attribute value changes.

**Significance**
Previously, separate code had to be written to handle actions corresponding to changes in pov-3d-viewer's attributes. This enhancement resolves this issue and enables pov-3d-viewer to automatically adapt to attribute value changes.

**Operation Method**
observedAttributes: This method returns the list of attributes that pov-3d-viewer will observe. Currently, it is configured to observe the asset, preset, and base_color attributes.
attributeChangedCallback: This method is called when the value of an observed attribute changes. It receives the attribute's name, its previous value, and its new value as arguments, allowing for the definition of actions based on attribute value changes. For example, when the asset attribute's value changes, a new 3D model is loaded, and when the base_color attribute's value changes, the default color of the 3D model is updated.

<br/>

## Types of Changes

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: examples/viewer_example_in_html/example.html
Comment (jack-kdl) [inline examples/viewer_example_in_html/example.html:1]: vanila js example이군요. 명확하게 example_javascript.html 은 어떨까요?

File: examples/viewer_example_in_html/example.html [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      <div class="description">
+        <header>
+          <h1><a href="../index.html">Pov-3d</a></h1>
+        </header>
+        <p>
+          This is a simple example of how to use Pov-3d. You can drag and drop a
+          glTF 2.0 file or folder here to view it.
+        </p>
+      </div>
+      <div class="my_viewer"></div>
```
Comment (jack-kdl) [inline examples/viewer_example_in_html/example.html]: example인 만큼 명확하게 id로 delegation 하는건 어떨까요?

File: src/viewer.js [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+const THREE_PATH = `https://unpkg.com/three@0.${REVISION}.x`;
+
+const DRACO_LOADER = new DRACOLoader(MANAGER).setDecoderPath(
+  `${THREE_PATH}/examples/jsm/libs/draco/gltf/`,
+);
+const KTX2_LOADER = new KTX2Loader(MANAGER).setTranscoderPath(
+  `${THREE_PATH}/examples/jsm/libs/basis/`,
+);
+
+class Viewer {
```
Comment (jack-kdl) [inline src/viewer.js]: 클래스 이름이 좀 너무 제너럴 하지 않나요?

File: examples/viewer_example_in_html/example.html [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <link rel="stylesheet" href="../example.style.css" />
+    <title>Pov-3d Example</title>
+    <script src="../../vite.config.js"></script>
+  </head>
+  <body>
+    <script type="module">
+      import Viewer from "../../build/viewer";
```
Comment (jack-kdl) [inline examples/viewer_example_in_html/example.html]: 요 방법은 제너럴 한건가요? 브라우저 호환성이 떨어질꺼 같은데?

<script src="https://.../pov-3d.js"></script>
요 방법이 좀더 제너럴하고 브라우저 호환성도 지킬수있지 않나요?

File: examples/viewer_example_in_html/example.html [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+      document.addEventListener("DOMContentLoaded", async () => {
+        const wrapper = document.querySelector(".my_viewer");
+        if (!wrapper) {
+          throw new Error("Wrapper not found");
+        }
+
+        const exampleViewer = new Viewer(wrapper);
+        await exampleViewer.load("../sampleAssets/Old_ToadCenser.glb");
+      });
```
Comment (jack-kdl) [inline examples/viewer_example_in_html/example.html]: 개발자들이 DOMContentLoaded 이벤트를 알아야할까요?

pov_3d_viewer('#my_viewer', (viewer) => {
    viewer.load('....');
});
이런형태는 어떨까요?

pov_3d_viewer('.my_viewer', ...);
pov_3d_viewer(document.querySelector(".my_viewer"), ...);

이렇게 사용자에게 다양성을 주는것도 좋을듯하고요.

File: examples/viewer_example_in_js/example.js
Comment (jack-kdl) [inline examples/viewer_example_in_js/example.js:1]: 위에 example.html 하고 뭐가 다르죠?

File: src/viewer.js [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+import { DRACOLoader } from "three/examples/jsm/loaders/DRACOLoader.js";
+import { MeshoptDecoder } from "three/examples/jsm/libs/meshopt_decoder.module.js";
+import { KTX2Loader } from "three/examples/jsm/loaders/KTX2Loader.js";
+import { RoomEnvironment } from "three/examples/jsm/environments/RoomEnvironment.js";
+import { EXRLoader } from "three/examples/jsm/loaders/EXRLoader.js";
+
+const MANAGER = new LoadingManager();
+const THREE_PATH = `https://unpkg.com/three@0.${REVISION}.x`;
+
+const DRACO_LOADER = new DRACOLoader(MANAGER).setDecoderPath(
```
Comment (jack-kdl) [inline src/viewer.js]: 잉? main module에서 example을 reference?

File: styles.css
Comment (jack-kdl) [inline styles.css:1]: 이파일 용도는 뭔데 여기 위치하나요?

**Review by jack-kdl — CHANGES_REQUESTED** [review]
우리는 다양성을 가지는 솔루션을 제공하고자 함으로 사용성과 일관성을 조금더 잡고 가시죠

File: src/viewer.js (hunk 5 above)
Comment (JOHNKIM-KK) [reply src/viewer.js]: three 의 example을 경로를 의미합니다!
저희 Pov-3d의 example는 reference하지 않았습니다!

File: styles.css
Comment (JOHNKIM-KK) [reply styles.css:1]: 프로젝트 초기세팅시 있던 style.css입니다.
현재 사용하고 있지 않아서 제거 하였습니다.

File: src/pov_3d_viewer.js [hunk 6]
```diff
@@ -0,0 +1,495 @@
...
+import { GLTFLoader } from "three/addons/loaders/GLTFLoader.js";
+import { OrbitControls } from "three/addons/controls/OrbitControls.js";
+import { FBXLoader } from "three/examples/jsm/loaders/FBXLoader.js";
+import { DRACOLoader } from "three/examples/jsm/loaders/DRACOLoader.js";
+import { MeshoptDecoder } from "three/examples/jsm/libs/meshopt_decoder.module.js";
+import { KTX2Loader } from "three/examples/jsm/loaders/KTX2Loader.js";
+import { RoomEnvironment } from "three/examples/jsm/environments/RoomEnvironment.js";
+import { EXRLoader } from "three/examples/jsm/loaders/EXRLoader.js";
+import ViewerOption from "./option/viewerOption";
+import { OBJLoader } from "three/examples/jsm/loaders/OBJLoader.js";
```
Comment (kimswon) [inline src/pov_3d_viewer.js]: do not import just example, import logic

File: src/pov_3d_viewer.js [hunk 7]
```diff
@@ -0,0 +1,495 @@
...
+
+const MANAGER = new LoadingManager();
+const THREE_PATH = `https://unpkg.com/three@0.${REVISION}.x`;
+
+const DRACO_LOADER = new DRACOLoader(MANAGER).setDecoderPath(
```
Comment (kimswon) [inline src/pov_3d_viewer.js:25]: treejs treat loader as a adopter, inherit addon mechanism on our solution

File: src/pov_3d_viewer.js [hunk 8]
```diff
@@ -0,0 +1,495 @@
...
+        detail: {
+          type: "load-progress",
+          loaded: progressEvent.loaded,
+          total: progressEvent.total,
+        },
+      }),
+    );
+  };
+
+  async load(file) {
```
Comment (kimswon) [inline src/pov_3d_viewer.js:321]: make it as loader system

**Review by kimswon — CHANGES_REQUESTED** [review]

**Pull request merged**